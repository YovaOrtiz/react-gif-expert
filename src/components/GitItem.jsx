import PropTypes from 'prop-types';
import React from 'react';


export const GitItem = ({ title, url, id }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GitItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,

};
