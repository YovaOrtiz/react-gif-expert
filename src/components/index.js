
export * from'./AddCategory';
export * from'./GitGrind';
export * from'./GitItem';
