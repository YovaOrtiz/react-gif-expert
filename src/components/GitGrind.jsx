import { GitItem } from "./GitItem";
import { useFetchGrid } from "../hooks/useFetchGrid";


export const GitGrind = ({category}) => {

  const {images, isLoading} = useFetchGrid (category);

  return (
    <>
       <h3>{category}</h3>
       {
        isLoading && (<h2>cargando...</h2>)
       }

       <div className="card-grid">
        {
          images.map( (image) => (
            <GitItem 
            key={image.id}
            {...image}
            />

          ))
        }

       </div>
  
     
    </>
  )
}
