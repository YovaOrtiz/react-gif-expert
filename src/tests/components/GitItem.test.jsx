import React from 'react';
import { render, screen } from "@testing-library/react";
import { GitItem } from "../../components/GitItem";

describe('Pruebas en <GitItem/>', () => {  

    const title ='Saitana';
    const url = 'https://one-punch.com/saitana.jpg';

    test('debe de hacer match con el snapshot', () => {
        
        const {container} =  render(<GitItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();

    });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GitItem title={test} url={url}/>);
        //screen.debug();
        //expect(screen.getByRole('img').src);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt)

        
    });

    test('debe de mostrar el titulo en el componente', () => {
        render(<GitItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();
    });


});