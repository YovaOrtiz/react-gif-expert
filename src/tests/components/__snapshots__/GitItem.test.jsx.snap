// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GitItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitana"
      src="https://one-punch.com/saitana.jpg"
    />
    <p>
      Saitana
    </p>
  </div>
</div>
`;
